lib/rbac/filterer.rb

```diff
@@ -223,7 +223,9 @@ def search(options = {})
       end
 
       if search_filter && targets && (!exp_attrs || !exp_attrs[:supported_by_sql])
-        rejects     = targets.reject { |obj| matches_search_filters?(obj, search_filter, tz) }
+        rejects = TimeProfile.with_cached_time_profile_ids do
+          targets.reject { |obj| matches_search_filters?(obj, search_filter, tz) }
```

Comment:
as @NickLaMuro suggested, it's odd that we're talking about caching time profile ids in the `matches_search_filters?` section... We're not sure if this caching, if needed after other PRs are merged, should be at this layer.  🤔